jdaviz/configs/specviz2d/plugins/cross_dispersion_profile/cross_dispersion_profile.py

```diff
@@ -0,0 +1,313 @@
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._marks = {}
+
+        # description displayed under plugin title in tray
+        self._plugin_description = 'Visualize cross-dispersion profile.'
+
+        self.dataset = DatasetSelect(self,
+                                     'dataset_items',
+                                     'dataset_selected',
+                                     filters=['layer_in_spectrum_2d_viewer',
+                                              'not_trace'])
+
+        self.hub.subscribe(self, GlobalDisplayUnitChanged,
+                           handler=self._on_display_units_changed)
+
+        # attribute to access computed profile, will be a quantity array
+        self._profile = None
+
+        # override default plot styling
+        self.plot.figure.fig_margin = {'top': 60, 'bottom': 60, 'left': 65,
+                                       'right': 15}
+        self.plot.viewer.axis_y.tick_format = '0.1e'
+        self.plot.viewer.axis_y.label_offset = '50px'
+
+    @property
+    def user_api(self):
+        expose = ('dataset', 'pixel', 'y_pixel', 'use_full_width', 'width',
+                  'profile')
+        return PluginUserApi(self, expose=expose)
+
+    @observe("dataset_selected")
+    def _set_defaults(self, event={}):
+        """
+        When a dataset is selected, re-calculate the default values for pixel,
+        y_pixel, width, and the slider limits for selecting row/column where
+        the profile will be measured.
+        """
+        # self.dataset might not exist when app is setting itself up.
+        if hasattr(self, "dataset") and self.dataset.selected_obj is not None:
+            data = self.dataset.selected_obj
+            # default value for 'y_pixel' is middle of cross dispersion axis
+            self.y_pixel = math.floor(data.shape[0] / 2)
+            # default value for 'pixel' is middle of spectral axis
+            self.pixel = math.floor(data.shape[1] / 2)
+            self._pixel_to_wav()  # also set corresponding wavelength
+            # slider limits
+            self.max_y_pix = data.shape[0]
+            self.max_pix = data.shape[1]
+            # default use_full_width=True
+            self.use_full_width = True
+            # set appropriate default 'width' if use_full_width=False
+            self.width = data.shape[0]
+
+    @observe('pixel', 'sa_display_unit')
+    def _pixel_to_wav(self, event={}):
+        """
+        Calculate the corresponding wavelength for ``pixel``, if wcs is present,
+        when ``pixel`` is changed.
+        """
+        data = self.dataset.selected_obj
+        if data is not None:
+            if hasattr(data, 'wcs') and self.sa_display_unit != '':
+                wcs = self.dataset.selected_obj.wcs
+                wav = wcs.pixel_to_world(self.pixel)
+                self.wav = wav.to(u.Unit(self.sa_display_unit), u.spectral()).value

```

Comment:
does this need to react to a change in the data as well, perhaps from data with to without wcs?